sdk/test-utils/recorder/src/utils.ts

```diff
@@ -354,3 +360,85 @@ export function findRecordingsFolderPath(filePath: string): string {
     );
   }
 }
+
+export function formatPath(path: string): string {
+  return path
+    .toLowerCase()
+    .replace(/ /g, "_")
+    .replace(/<=/g, "lte")
+    .replace(/>=/g, "gte")
+    .replace(/</g, "lt")
+    .replace(/>/g, "gt")
+    .replace(/=/g, "eq")
+    .replace(/\W/g, "");
+}
+
+/**
+ * Generates a file path with the following structure:
+ *
+ *     `{node|browsers}/<describe-block-title>/recording_<test-title>.{js|json}`
+ *
+ * @param platform A string, either "node" or "browsers".
+ * @param testSuiteTitle The title of the test suite.
+ * @param testTitle The title of the specific test we're running.
+ */
+export function generateTestRecordingFilePath(
+  platform: "node" | "browsers",
+  testSuiteTitle: string,
+  testTitle: string
+): string {
+  // File Extension
+  // nock recordings for node tests - .js extension
+  // recordings are saved in json format for browser tests - .json extension
+  const ext = platform === "node" ? "js" : "json";
+  return `${platform}/${formatPath(testSuiteTitle)}/recording_${formatPath(testTitle)}.${ext}`;
+}
+
+/**
+ * Requires a file if it exists. Only works on NodeJS.
+ */
+export function nodeRequireRecordingIfExists(recordingPath: string, testAbsolutePath: string): any {
+  if (isBrowser()) throw new Error("nodeRequireRecordingIfExists only works on NodeJS");
+  const path = require("path");
+  // Get the full path of the `recordings` folder by navigating through the hierarchy of the test file path.
+  const recordingsFolderPath = findRecordingsFolderPath(testAbsolutePath);
+  const absoluteRecordingPath = path.resolve(recordingsFolderPath, recordingPath);
+  if (fs.existsSync(absoluteRecordingPath)) {
+    return require(absoluteRecordingPath);
+  } else {
+    throw new Error(`The recording ${recordingPath} was not found in ${recordingsFolderPath}`);
+  }
+}
+
+/**
+ * Checks if a test hasn't changed from the last time it was recorded.
+ * @param testContext
+ * @param testSuiteTitle
+ * @param testTitle
+ * @param currentHash
+ */
+export function testHasChanged(
+  testSuiteTitle: string,
+  testTitle: string,
+  testAbsolutePath: string,
+  currentHash: string
+): boolean {
+  const platform = isBrowser() ? "browsers" : "node";
+  const recordingPath: string = generateTestRecordingFilePath(platform, testSuiteTitle, testTitle);
+
+  let previousHash: string = "";
+
+  if (platform === "node") {
+    try {
+      previousHash = nodeRequireRecordingIfExists(recordingPath, testAbsolutePath).hash;
+    } catch (e) {}
+  } else {
+    previousHash = (window as any).__json__["recordings/" + recordingPath].hash;
```

Comment:
let's do it this way by handling the case where the previous hash is not present, consider that the hash has been changed.